staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go

```diff
@@ -471,31 +477,184 @@ func TestUpdateNodesInExternalLoadBalancer(t *testing.T) {
 				{Service: newService("s1", "888", v1.ServiceTypeLoadBalancer), Hosts: nodes},
 				{Service: newService("s3", "999", v1.ServiceTypeLoadBalancer), Hosts: nodes},
 			},
+			worker: 5,
 		},
 		{
-			// One service has an external load balancer and one is nil: one call.
+			desc: "One service has an external load balancer and one is nil: one call.",
 			services: []*v1.Service{
 				newService("s0", "234", v1.ServiceTypeLoadBalancer),
 				nil,
 			},
 			expectedUpdateCalls: []fakecloud.UpdateBalancerCall{
 				{Service: newService("s0", "234", v1.ServiceTypeLoadBalancer), Hosts: nodes},
 			},
+			worker: 6,
+		},
+		{
+			desc: "Four services have external load balancer with only 2 workers",
+			services: []*v1.Service{
+				newService("s0", "777", v1.ServiceTypeLoadBalancer),
+				newService("s1", "888", v1.ServiceTypeLoadBalancer),
+				newService("s3", "999", v1.ServiceTypeLoadBalancer),
+				newService("s4", "123", v1.ServiceTypeLoadBalancer),
+			},
+			expectedUpdateCalls: []fakecloud.UpdateBalancerCall{
+				{Service: newService("s0", "777", v1.ServiceTypeLoadBalancer), Hosts: nodes},
+				{Service: newService("s1", "888", v1.ServiceTypeLoadBalancer), Hosts: nodes},
+				{Service: newService("s3", "999", v1.ServiceTypeLoadBalancer), Hosts: nodes},
+				{Service: newService("s4", "123", v1.ServiceTypeLoadBalancer), Hosts: nodes},
+			},
+			worker: 2,
 		},
 	}
 	for _, item := range table {
-		controller, cloud, _ := newController()
+		t.Run(item.desc, func(t *testing.T) {
+			controller, cloud, _ := newController()
+			controller.nodeLister = newFakeNodeLister(nil, nodes...)
+
+			var services []*v1.Service
+			services = append(services, item.services...)
```

Comment:
why this is needed?

Can't you just call updateLoadbalancerHosts(item.services, item.workers) below?